controllers/capa_controller.go

```diff
@@ -63,29 +64,22 @@ func (r *CAPAClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	var err error
 	logger := r.Log.WithValues("namespace", req.Namespace, "cluster", req.Name)
 
+	logger.Info("Reconciling AWSCluster")
+
 	cluster := &capa.AWSCluster{}
 	if err := r.Get(ctx, req.NamespacedName, cluster); err != nil {
-		logger.Info("Cluster no longer exists")
+		if k8sErrors.IsNotFound(err) {
+			logger.Info("Cluster no longer exists")
+			return ctrl.Result{}, nil
+		}
+
 		return ctrl.Result{}, microerror.Mask(err)
 	}
 
 	awsClusterRoleIdentity := &capa.AWSClusterRoleIdentity{}
-	err = r.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: req.Name}, awsClusterRoleIdentity)
+	err = r.Get(ctx, types.NamespacedName{Name: cluster.Spec.IdentityRef.Name}, awsClusterRoleIdentity)
```

Comment:
Can we skip passing the namespace here? I thought we needed to pass it when looking for the `AWSClusterRoleIdentity`.

Reply:
It's not a namespaced object. See `kubectl api-resources`. This also works in `dns-operator-aws` and `aws-network-topology-operator` already, for instance.